doc/cli.rst

```diff
@@ -45,11 +45,62 @@ This again generates a Tensorboard event file ``evaluation/events.out.tfevents.*
+In the following the most relevant settings for running experiments with DeepQMC are discussed.
 
-    $ deepqmc task.restdir=workdir ansatz=paulinet ansatz.omni_factory.gnn_factory.n_interactions=2
+Task
+____
 
-For convenience the configuration of the default ansatz is reproduced here:
+DeepQMC provides the above mentioned conigurations for the ``train``, ``evaluate`` and ``restart`` task. In order to override default hyperparameters of the experimental setup, such as the ``sample_size`` or the number of training ``steps`` or ``pretrain_steps``, hydra provides a simple syntax::
 
-.. literalinclude:: defaults.yaml
+        $ deepqmc task=train task.sample_size=2048 task.steps=50_000 task.pretrain_steps
+
+The working directory for logging and checkpointing is is defined through::
+
+        $ deepqmc hydra.run.dir=workdir
+
+Note that the working directory of an ``evaluate`` and ``restart`` task cannot match the ``restdir``.
+
+Hamiltonian
+___________
+
+DeepQMC aims at solving the molecular Hamiltonian. Molecules can be selected from a range of predefined configurations located in ``.../deepqmc/src/deepqmc/conf/ansatz``::
+
+        $ deepqmc hamil/mol=LiH
+
+The predefined configurations can be extended with custom molecules. Alternatively molecules can be specified on the command line::
+
+        $ deepqmc hamil.mol.coords=[[0,0,0],[0.742,0,0]] hamil.mol.charges=[1,1] hamil.mol.charge=0 hamil.mol.spin=0 hamil.mol.unit=angstrom
+
+Furthermore, DeepQMC implements the option to use pseudopotentials, which can be used via::
+
+        $ deepqmc hamil.mol.coords=[[0,0,0]] hamil.mol.charges=[20] hamil.mol.charge=0 hamil.mol.spin=0 hamil.mol.unit=angstrom +hamil.mol.pp_type='ccECP'
+
+Sampling
+________
+
+Different sampler configurations can be found in ``.../deepqmc/src/deepqmc/conf/ansatz``. A typical usecase would be to pick as sampler form these configurations and , if required, change some argument from the command line::
+
+        $deepqmc task/sampler=decorr_langevin task.sampler.0.length=30
+
+Optimization
+____________
+
+For the optimization either `KFAC <https://kfac-jax.readthedocs.io/en/latest/>`_ or optimizers from `optax <https://optax.readthedocs.io/en/latest/>`_ may be used. While the use of `KFAC <https://kfac-jax.readthedocs.io/en/latest/>`_ is highly recommended due to the significantly imporved convergence, at times it can be useful to run with other optimizers such as `AdamW <https://optax.readthedocs.io/en/latest/api.html#adamw>`_::
+
+        $ deepqmc task/opt=adamw
+
+Ansatz
+______
+
+The hyperparameters of the training and the wave function ansatz are specified through hydra config files. Predefined ansatzes can be found in ``.../deepqmc/src/deepqmc/conf/ansatz`` and selected via::
+
+    $ deepqmc ansatz=ferminet
+
+The hyperparameters of such a predifined ansatz can also be overwritten at the command line::
+
+    $ deepqmc ansatz.omni_factory.gnn_factory.n_interactions=2
```

Comment:
```suggestion
    $ deepqmc ansatz=ferminet ansatz.omni_factory.gnn_factory.n_interactions=2
```
It might be clearer if we continue with the previous example.

Reply:
My incentive was to reduce the complexity of these commands, but I don't mind changing it back if you prefer that.